### PR TITLE
Added UC21 - Get total game time for team and UC22 - Get max streak days played by team

### PR DESCRIPTION
**Pull request by henrymcarvajal**
Your comments, pls

### CODE REVIEW

File: app/src/db/team.js [hunk 1]
```diff
@@ -1,5 +1,14 @@
 const pool = require('./pool');
 
+const GET_TOTAL_GAME_TIME_FOR_TEAM_STATEMENT =
+    `SELECT coalesce(sum(gt.finish_time - gt.start_time),interval '0S') as total
```
Comment (FernandoArevaloE) [inline app/src/db/team.js:4]: what does `coalesce` do? I have never used it

Comment (henrymcarvajal) [reply app/src/db/team.js:4]: Returns the first not null argument from the list of arguments

File: app/init.sql [hunk 2]
```diff
@@ -87,6 +87,20 @@ CREATE TABLE prize (
     CONSTRAINT prize_pk PRIMARY KEY (prize_id)
 );
 
+CREATE TABLE public.game_time (
```
Comment (andymotagl) [inline app/init.sql:90]: I think this should be part of fame table even whether it does not exist yet

File: app/src/db/team.js [hunk 3]
```diff
@@ -1,5 +1,14 @@
...
+     FROM team t
+              JOIN player p on p.team_id = t.team_id
+              JOIN game_time gt ON gt.player_id = p.player_id
```
Comment (andymotagl) [inline app/src/db/team.js:7]: The statistics looks great. They must be for all teams tho

File: app/src/db/team.js [hunk 4]
```diff
@@ -1,5 +1,14 @@
...
+     WHERE t.team_id = $1
+       and gt.start_time >= $2
```
Comment (andymotagl) [inline app/src/db/team.js:9]: I guess it does not require this with all the games finished would be enough or if I'm wrong can you elaborate why this?

File: app/src/service/team.js [hunk 5]
```diff
@@ -159,10 +164,34 @@ const generateInvitationToken = async (teamId, actorId) => {
...
-exports.update = update;
-exports.getTeamJackpot = getTeamJackpot;
-exports.getAllTeamJackpot = getAllTeamJackpot;
-exports.generateInvitationToken = generateInvitationToken;
\ No newline at end of file
+const getStatistics = async (actorId, teamId, statisticName, statisticParams) => {
+    if (!teamId || !uuidTool.validate(teamId)) {
+        throw new BadRequest(`Invalid team id`);
+    }
+    if (!actorId || !uuidTool.validate(actorId)) {
```
Comment (andymotagl) [inline app/src/service/team.js:171]: It is not the leader who check this but the pluma blanca

File: app/src/service/team.js [hunk 6]
```diff
@@ -159,10 +164,34 @@ const generateInvitationToken = async (teamId, actorId) => {
...
+        throw new BadRequest('Invalid leader id ');
+    }
+    const team = await teamDAO.getTeamById(teamId)
```
Comment (andymotagl) [inline app/src/service/team.js:174]: It is for all teams

File: app/src/db/team.js (hunk 3 above)
Comment (henrymcarvajal) [reply app/src/db/team.js:7]: Fixed

File: app/src/service/team.js (hunk 5 above)
Comment (henrymcarvajal) [reply app/src/service/team.js:171]: Fixed

File: app/src/service/team.js (hunk 6 above)
Comment (henrymcarvajal) [reply app/src/service/team.js:174]: Fixed

File: app/src/db/team.js (hunk 4 above)
Comment (henrymcarvajal) [reply app/src/db/team.js:9]: Fixed

File: app/init.sql (hunk 2 above)
Comment (henrymcarvajal) [reply app/init.sql:90]: Will erge with developments by Fercho

**Review by andymotagl — APPROVED** [review]

File: app/src/db/stats/maxStreakDaysPerTeam.js [hunk 7]
```diff
@@ -0,0 +1,33 @@
+const SELECT_TEAM_CONSECUTIVE_DAYS =
+    `SELECT t.team_id,
+            gt.player_id,
+            gt.start_time,
+            ROW_NUMBER() OVER (PARTITION BY t.team_id ORDER BY gt.start_time) AS row_num,
+             gt.start_time - LAG(gt.start_time) OVER (PARTITION BY gt.player_id ORDER BY gt.start_time) AS day_diff
+      FROM team t
+      JOIN player p on p.team_id = t.team_id
+      JOIN game_time gt ON gt.player_id = p.player_id
```
Comment (FernandoArevaloE) [inline app/src/db/stats/maxStreakDaysPerTeam.js:9]: change game_time to game

Comment (FernandoArevaloE) [reply app/src/db/stats/maxStreakDaysPerTeam.js:9]: columns in game table are started and ended

Comment (henrymcarvajal) [reply app/src/db/stats/maxStreakDaysPerTeam.js:9]: Fixed!

**Review by FernandoArevaloE — APPROVED** [review]

**Pull request merged**